lib/api/CInferenceModelDefinition.cc

```diff
@@ -113,6 +128,29 @@ void addJsonArray(const std::string& tag,
     }
     writer.addMember(tag, array, parentObject);
 }
+
+template<typename T>
+void addJsonArray(const std::string& tag,
+                  const std::vector<T>& vector,
+                  CSerializableToJson::TGenericLineWriter& writer) {
+    writer.Key(tag);
+    rapidjson::Value array{writer.makeArray(vector.size())};
+    for (const auto& value : vector) {
+        array.PushBack(toJson(value), writer.getRawAllocator());
+    }
+    writer.write(array);
+}
```

Comment:
Although it seems nice to have this generic, are we actually only using it for `double` at the moment?  If so it would be more efficient to have the specific method:

```suggestion
void addJsonArray(const std::string& tag,
                  const std::vector<double>& vector,
                  CSerializableToJson::TGenericLineWriter& writer) {
    writer.Key(tag);
    writer.StartArray();
    for (const auto& value : vector) {
        writer.Double(value);
    }
    writer.EndArray();
}
```

If we _are_ writing different types of vectors then you can still avoid using the allocator by keeping this method a template but replacing `writer.Double(value);` in my suggestion with `writeSingleJsonValue(value);`, and then having multiple overloads of `writeSingleJsonValue` for all the types we need to write.

Reply:
Indeed, I have only a version supporting `doubles` now.